src/systems/filecoin_mining/storage_mining/_index.md

```diff
@@ -19,36 +19,44 @@ The Filecoin Storage Mining Subsystem ensures a storage miner can effectively co
 
-A sector that is in the `Failing` state for three consecutive `Proving Period`s will be terminated (`Terminated Faults`) meaning its data will be deemed unrecoverable by PoSt and the sector will have to be SEALed once more (as part of PoRep).
+{{< diagram src="diagrams/sector_state_machine.dot.svg" title="Sector State Machine" >}}
 
-Conversely `RecoverFaults()` can be called any time by the miner on a failing sector to return it to the `ProvingSet` and attempt to prove data is being stored once more. For instance an Active sector might move into the failing state during a power outage (through a declared or detected fault). At the end of the outage, the miner may call `RecoverFaults` to transition the state to `Recovering` before proving it once more and returning it to `Active`. 
+{{< diagram src="diagrams/sector_state_machine_legend.dot.svg" title="Sector State Machine Legend" >}}
 
-We illustrate these states below.
+### Miner PoSt State
 
-{{< readfile file="storage_mining_subsystem.id" code="true" lang="go" >}}
+`MinerPoStState` keeps track of a miner's state in responding to PoSt and there are three states in `MinerPoStState`:
 
-### Sector in StorageMiner State Machine (new one)
+- `OK` miner has passed either a ElectionPoSt or a SurprisePoSt sufficiently recently.
+- `Challenged` miner has been selected to prove its storage via SurprisePoSt and is currently in the Challenged state
+- `DetectedFault` miner has failed at least one SurprisePoSt, indicating that all claimed storage may not be proven. Miner has lost power on its sector and recovery can only proceed by a successful response to a subsequent SurprisePoSt challenge, up until the limit of number of consecutive failures.
 
-{{< diagram src="diagrams/sector_state_fsm.dot.svg" title="Sector State (new one)" >}}
+`DetectedFault` is a miner-wide PoSt state when all sectors are considered inactive. All power is lost immediately and pledge collateral is slashed. If a miner remains in `DetectedFault` for more than MaxConsecutiveFailures, all sectors will be terminated, both power and market actors will be notified.
 
-{{< diagram src="diagrams/sector_state_legend.dot.svg" title="Sector State Legend (new one)" >}}
+`ProvingSet` is consist of sectors that miners are required to generate proofs against and is what counts towards miners' power. In other words, `ProvingSet` is a set of all `Active` sectors for a particular miner. `ProvingSet` is only relevant when the miner is in OK stage of its `MinerPoStState`. When a miner is in the `Challenged` state, `ChallengedSectors` specify the list of sectors to be challenged which is the `ProvingSet` before the challenge is issued thus allowing more sectors to be added while it is in the `Challenged` state.
```

Comment:
- typo s/is consist/consists
This sentence: 
> When a miner is in the `Challenged` state, ...

is a bit confusing. would we want to allow miners to add more sectors while in the challenged state? And what of detected vs declared faults?

Reply:
typo has been fixed.

per second comment, we will still want to allow miners to add more sectors while in challenged state (this is a requirement from impl). I think this wont affect Challenged state since it specifies a list of ChallengedSectors. Miners should not be allowed to declare fault when it is in DetectedFault state. I have flagged this to @jzimmerman and think this deserves a bit more discussion. We can come back to this in a separate PR.

Reply:
I agree with all your statements here. I think in order to ensure faults are declared appropriately (and to incentivize it) we should default to what you said (ie cant declare fault if already detected). I would add now (+ in code) but understand deferral.